contracts/attesters/hydra-s1/variants/HydraS1AccountboundAttester.sol

```diff
@@ -165,30 +119,32 @@ contract HydraS1AccountboundAttester is IHydraS1AccountboundAttester, HydraS1Bas
   *******************************************************/
   /**
    * @dev Hook run before recording the attestation.
-   * Throws if ticket already used, not a renewal, and ticket on cooldown.
+   * Throws if nullifier already used, not a renewal, and nullifier on cooldown.
    * @param request users request. Claim of having an account part of a group of accounts
    * @param proofData provided to back the request. snark input and snark proof
    */
-  function _beforeRecordAttestations(Request calldata request, bytes calldata proofData)
-    internal
-    virtual
-    override
-  {
-    uint256 userTicket = proofData._getTicket();
-    TicketData memory userTicketData = _userTicketsData[userTicket];
+  function _beforeRecordAttestations(
+    Request calldata request,
+    bytes calldata proofData
+  ) internal virtual override {
+    uint256 nullifier = proofData._getNullifier();
+    address nullifierDestination = _getDestinationOfNullifier(nullifier);
+
+    HydraS1Claim memory claim = request._claim();
 
     if (
-      userTicketData.destination != address(0) && userTicketData.destination != request.destination
+      _isCooldownRenewedForNullifier(
```

Comment:
This is a "is" function that should be true/false and it has a revert inside, i think it's a little bit weird to have it there